views/middleware.go

```diff
@@ -48,16 +50,134 @@ func (v *Views) RequiresPermission(h http.Handler, p permission.Permission) http
 			return
 		}
 
+		acceptedPerms := GetValidPermissions(p)
+
 		for _, perm := range perms {
-			if perm.Name == p.Name {
+			if acceptedPerms[perm] {
 				h.ServeHTTP(w, r)
 				return
 			}
 		}
-		err = v.template.RenderNoNavsTemplate(w, nil, templates.ForbiddenTemplate)
+
+		err = v.template.RenderNoNavsTemplate(w, nil, templates.Forbidden500Template)
 		if err != nil {
 			log.Println(err)
 		}
 		w.WriteHeader(http.StatusForbidden)
 	}
 }
+
+func (v *Views) RequiresMinimumPermissionNoHttp(userID int, p permissions.Permissions) bool {
+	u, err := v.user.GetUser(context.Background(), user.User{UserID: userID})
+	if err != nil {
+		log.Println(err)
+		return false
+	}
+
+	p1, err := v.user.GetPermissionsForUser(context.Background(), u)
+	if err != nil {
+		log.Println(err)
+		return false
+	}
+
+	m := GetValidPermissions(p)
+
+	for _, perm := range p1 {
+		if m[perm] {
+			return true
+		}
+	}
+
+	return false
+}
+
+func GetValidPermissions(p permissions.Permissions) (m map[string]bool) {
+	m = make(map[string]bool)
+	if p.GetString() == permissions.MenuDisabled.GetString() {
+		m[p.GetString()] = true
+		return
+	}
+
+	m[p.GetString()] = true
+
+	switch p {
+	case permissions.ManageMembersAdmin:
+	case permissions.KeyCardAccess:
+	case permissions.BookingsAdmin:
+	case permissions.CalendarAdmin:
+	case permissions.CMSAdmin:
+	case permissions.Cobra:
+	case permissions.Director:
+	case permissions.EditNetStats:
+	case permissions.EmailEveryone:
+	case permissions.EquipmentAdmin:
+	case permissions.HiresAdmin:
+	case permissions.Inform:
+	case permissions.KeyListManage:
+	case permissions.MailingListAdmin:
+	case permissions.OfficerReports:
+	case permissions.Streamer:
+	case permissions.TechieTodo:
+	case permissions.VideoStats:
+	case permissions.WatchAdmin:
```

Comment:
I think this is at least the second time I see a "switch on all permissions" in this PR - could it be refactored out?

Reply:
Fixed, was like this before because of import cycles